src/scanner.rs

```diff
@@ -619,6 +624,14 @@ mod tests {
         );
     }
 
+    #[test]
+    fn it_errors_on_invalid_utf8_char_for_issue_6() {
+        assert_has_error(
+            "\"\\uDF06\"",
+            "Invalid unicode escape sequence. 'DF06' is not a valid UTF8 character on line 1 column 2.",
+        );
+    }
```

Comment:
@lucacasonato do you think this is ok behaviour?

Reply:
This is good, but having it be represented would be even better.

Reply:
Opened #8 for later. Going to merge this for now.